x-pack/plugins/ml/public/embeddables/anomaly_charts/anomaly_charts_embeddable_factory.tsx

```diff
@@ -0,0 +1,223 @@
+      }
+      const [coreStartServices, pluginsStartServices] = await getStartServices();
+      const anomalyChartsDependencies = await getAnomalyChartsServiceDependencies(
+        coreStartServices,
+        pluginsStartServices
+      );
+      const [, , mlServices] = anomalyChartsDependencies;
+
+      const subscriptions = new Subscription();
+
+      const { titlesApi, titleComparators, serializeTitles } = initializeTitles(state);
+      const {
+        api: timeRangeApi,
+        comparators: timeRangeComparators,
+        serialize: serializeTimeRange,
+      } = initializeTimeRange(state);
+
+      const {
+        anomalyChartsControlsApi,
+        dataLoadingApi,
+        serializeAnomalyChartsState,
+        anomalyChartsComparators,
+        onAnomalyChartsDestroy,
+      } = initializeAnomalyChartsControls(state, titlesApi, parentApi);
+
+      const api = buildApi(
+        {
+          isEditingEnabled: () => true,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.ml.components.mlAnomalyExplorerEmbeddable.typeDisplayName', {
+              defaultMessage: 'anomaly charts',
+            }),
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableAnomalyChartsUserInput } = await import(
+                './anomaly_charts_setup_flyout'
+              );
+              const result = await resolveEmbeddableAnomalyChartsUserInput(
+                coreStartServices,
+                pluginsStartServices,
+                parentApi,
+                uuid,
+                {
+                  ...serializeTitles(),
+                  ...serializeAnomalyChartsState(),
+                }
+              );
+              anomalyChartsControlsApi.updateUserInput(result);
+            } catch (e) {
+              // eslint-disable-next-line no-console
+              console.error(e);
+              return Promise.reject();
+            }
+          },
+          ...titlesApi,
+          ...timeRangeApi,
+          ...anomalyChartsControlsApi,
+          ...dataLoadingApi,
+          dataViews: buildDataViewPublishingApi(
+            {
+              anomalyDetectorService: mlServices.anomalyDetectorService,
+              dataViewsService: pluginsStartServices.data.dataViews,
+            },
+            { jobIds: anomalyChartsControlsApi.jobIds$ },
+            subscriptions
+          ),
+          serializeState: () => {
+            return {
+              rawState: {
+                timeRange: undefined,
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeAnomalyChartsState(),
+              },
+              references: [],
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...anomalyChartsComparators,
+        }
+      );
+
+      const appliedTimeRange$: Observable<TimeRange | undefined> = combineLatest({
```

Comment:
AFAIK for dashboard embeddable best practices - we should be using `fetch$` from `@kbn/presentation-publishing` to get the timeRange from the parent. 
cc @ThomThomson for confirmation

Reply:
Yes this is very true. Ideally `fetch$` can simplify this code quite a lot!

Reply:
Thanks for the hint! Updated here https://github.com/elastic/kibana/pull/183456/commits/1ed16a4c27a7d094f8c60beda5f13c7b95d58c7a